activity_duration/src/pulse_points.rs

```diff
@@ -0,0 +1,81 @@
+use std::time::Duration;
+
+use crate::{Activity, ActivityKind};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct PulseRecord {
+    pub duration: Duration,
+    pub category: PulseRateCategory,
+}
+
+impl From<(PulseRateCategory, Duration)> for PulseRecord {
+    fn from((category, duration): (PulseRateCategory, Duration)) -> Self {
+        Self { duration, category }
+    }
+}
+
+impl From<Activity> for PulseRecord {
+    fn from(Activity { kind, duration, .. }: Activity) -> Self {
+        Self {
+            duration,
+            category: kind.into(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum PulseRateCategory {
+    Low,
+    Medium,
+    High,
+}
+
+impl From<ActivityKind> for PulseRateCategory {
+    fn from(value: ActivityKind) -> Self {
+        match value {
+            ActivityKind::VO2 | ActivityKind::Anaerobic | ActivityKind::Aerobic => Self::High,
+            ActivityKind::FatBurn | ActivityKind::WarmUp => Self::Medium,
+            ActivityKind::Resting => Self::Low,
+        }
+    }
+}
+
+impl PulseRateCategory {
+    pub const fn weights(self) -> f64 {
+        match self {
+            PulseRateCategory::Low => 1.0,
+            PulseRateCategory::Medium => 1.5,
+            PulseRateCategory::High => 2.0,
```

Comment:
I suggest that you could use something like Fibonacci sequence for the coefficients. It is because the low level heart rate takes less energy than the high. The difference could be like 10 times. You could omit the comment for now.